### PR TITLE
expose docs for custom_vjp.defvjp

### PR DESCRIPTION
**Pull request by cgarciae**
# Changes

Uses `autoclass` for `custom_vjp` so `defvjp` is properly documented.

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 absl-py
 ipython>=8.8.0  # 8.7.0 has ipython3 lexer error
-sphinx>=7.3.2  # 7.3.0 breaks sphinx-book-theme
+sphinx~=7.4.0  # 7.3.0 breaks sphinx-book-theme
```
Comment (jakevdp) [inline docs/requirements.txt:3]: Let's make this more specific:
```suggestion
sphinx>=7.3.2,<8.0 # 7.3.0 breaks sphinx-book-theme; 8+ triggers warnings in myst_nb v1.1
```

Comment (jakevdp) [reply docs/requirements.txt:3]: I'm making this change separately in #22896 in order to unbreak doc builds at head.

Comment (cgarciae) [reply docs/requirements.txt]: Thanks @jakevdp! Rebased, testing...

**Review by jakevdp — COMMENTED** [review]
Looks great, thanks!

**Review by jakevdp — APPROVED** [review]

**Comment by dfm** [comment]
This doesn't look quite right to me! `custom_vjp` is now documented inline at https://jax--22895.org.readthedocs.build/en/22895/jax.html#jax.custom_vjp rather than on its own page. I don't think that's what we want, right?

**Comment by dfm** [comment]
I'm doing some housekeeping and I'm going to close this because I think #23069 did the trick. @cgarciae: please re-open if I'm missing something. Thanks!

**Pull request closed without merging**